pkg/controller/bastion/configvalidator.go

```diff
@@ -0,0 +1,127 @@
+		logger:           logger.WithName("gcp-bastion-config-validator"),
+	}
+}
+
+// Validate validates the provider config of the given bastion resource with the cloud provider.
+func (c *configValidator) Validate(ctx context.Context, bastion *extensionsv1alpha1.Bastion, cluster *extensions.Cluster) field.ErrorList {
+	allErrs := field.ErrorList{}
+
+	logger := c.logger.WithValues("bastion", client.ObjectKeyFromObject(bastion))
+
+	infrastructureStatus, err := getInfrastructureStatus(ctx, c.Client(), cluster)
+	if err != nil {
+		allErrs = append(allErrs, field.InternalError(nil, err))
+		return allErrs
+	}
+
+	secretReference := &corev1.SecretReference{
+		Namespace: cluster.ObjectMeta.Name,
+		Name:      v1beta1constants.SecretNameCloudProvider,
+	}
+
+	// Create GCP compute client
+	computeClient, err := c.gcpClientFactory.NewComputeClient(ctx, c.Client(), *secretReference)
+	if err != nil {
+		allErrs = append(allErrs, field.InternalError(nil, err))
+		return allErrs
+	}
+
+	// Validate bastion config
+	logger.Info("Validating bastion configuration")
+	allErrs = append(allErrs, c.validateInfrastructureStatus(ctx, computeClient, cluster.Shoot.Spec.Region, infrastructureStatus)...)
+
+	return allErrs
+}
+
+func getInfrastructureStatus(ctx context.Context, c client.Client, cluster *extensions.Cluster) (*gcp.InfrastructureStatus, error) {
+	var infrastructureStatus *gcp.InfrastructureStatus
+
+	worker := &extensionsv1alpha1.Worker{}
+
+	err := c.Get(ctx, client.ObjectKey{Namespace: cluster.ObjectMeta.Name, Name: cluster.Shoot.Name}, worker)
+	if err != nil {
+		return nil, err
+	}
+
+	if worker == nil || worker.Spec.InfrastructureProviderStatus == nil {
+		return nil, errors.New("infrastructure provider status must be not empty for worker")
+	}
+
+	if infrastructureStatus, err = helper.InfrastructureStatusFromRaw(worker.Spec.InfrastructureProviderStatus); err != nil {
+		return nil, err
+	}
+
+	if infrastructureStatus.Networks.VPC.Name == "" {
+		return nil, errors.New("virtual network must be not empty for infrastructure provider status")
+	}
+
+	if len(infrastructureStatus.Networks.Subnets) == 0 || infrastructureStatus.Networks.Subnets[0].Name == "" {
+		return nil, errors.New("subnet must be not empty for infrastructure provider status")
+	}
+
+	return infrastructureStatus, nil
+}
+
+func (c *configValidator) validateInfrastructureStatus(ctx context.Context, computeClient gcpclient.ComputeClient, region string, infrastructureStatus *gcp.InfrastructureStatus) field.ErrorList {
+	allErrs := field.ErrorList{}
+
+	vpc, err := computeClient.GetVPC(ctx, infrastructureStatus.Networks.VPC.Name)
+	if err != nil || vpc == nil || vpc.Name == "" {
+		allErrs = append(allErrs, field.InternalError(field.NewPath("vpc"), fmt.Errorf("could not get vpc %s from gcp provider: %w", infrastructureStatus.Networks.VPC.Name, err)))
+		return allErrs
+	}
+
+	subnet, err := computeClient.GetSubenet(ctx, region, infrastructureStatus.Networks.Subnets[0].Name)
```

Comment:
You shouldn't use the subnet by slice index, rather the subnet with purpose `nodes`. There isn't a guarantee that this is the first will be the worker subnet and not the internal one.

Reply:
Thanks, commit in https://github.com/gardener/gardener-extension-provider-gcp/pull/497/commits/ea08d3c63e779f44c1644aa3e6567abe3bab1c62 @kon-angelo